overcooked_ai_py/mdp/mdp_utils.py

```diff
@@ -0,0 +1,537 @@
+import random
+import copy
```

Comment:
Curious about where we use most of the utils defined in this file: is it for potentials and layout generation?

Reply:
Functions in this file was basically a very stripped down version of planner (with the only API to find distance and plans, also ignoring orientations). But instead of pre-computing plans, it only compute plans on request.

This was first used in my version of the potential function. Nathan, in his latest PR for overcooked_ai, uses the full planner directly in the potential function.

I think it makes sense for him to use the full planner since he will be using a handful of layout only. But for the multi-layout, it is hard to come across the same layout multiple times, so I am not sure if we should use the full planner.

Reply:
Yeah that makes sense. I think we should add some text about where this is used at the top of the file. And if this is the only location where it's used, we might want to call this file `lightweight_planning.py` or `planning_ulils.py` something like this (feel free to choose a better name).